### PR TITLE
fix: Types mentions.

### PR DESCRIPTION
**Pull request by kutuzov13**
# Checklist

- [ ] I've written good commit message for all commits
- [ ] I've split changes into separate commits where it's appropriate
- [ ] I've updated project version in `pyproject.toml`
- [ ] I'll make a release when PR is merged

### CODE REVIEW

File: pybotx/models/message/mentions.py [hunk 1]
```diff
@@ -16,52 +16,101 @@
...
+    def __str__(self) -> str:
+        return self._build_embed_mention(self.type)
+
+
+@dataclass
+class MentionAll:
+    type: Literal[MentionTypes.ALL]
+
+    def __str__(self) -> str:
+        return f"<embed_mention>{self.type.value}::</embed_mention>"  # noqa: WPS237
```
Comment (Nimond) [inline pybotx/models/message/mentions.py:66]: Also use `_build_embed_mention` here. May be pass `entity_id` and `name` through args?

File: pybotx/models/message/mentions.py [hunk 2]
```diff
@@ -16,52 +16,101 @@
 
 
 @dataclass
-class Mention:
-    type: MentionTypes
-    entity_id: Optional[UUID] = None
-    name: Optional[str] = None
+class BaseMentionEntity:
```
Comment (Nimond) [inline pybotx/models/message/mentions.py:19]: `BaseTargetMention`

**Review by Nimond — CHANGES_REQUESTED** [review]

File: pybotx/models/message/mentions.py (hunk 2 above)
Comment (kutuzov13) [reply pybotx/models/message/mentions.py]: Fix

File: pybotx/models/message/mentions.py (hunk 1 above)
Comment (kutuzov13) [reply pybotx/models/message/mentions.py]: Fix

File: pybotx/models/message/mentions.py [hunk 3]
```diff
@@ -15,53 +15,123 @@
...
+    name: Optional[str],
+) -> str:
+    name = name or ""
+    entity_id = entity or ""
+    return f"<embed_mention>{mention_type.value}:{entity_id}:{name}</embed_mention>"
+
+
+@dataclass
+class BaseMention:
+    str_entity_id: Optional[UUID]
```
Comment (Nimond) [inline pybotx/models/message/mentions.py:30]: `entity_id`

File: pybotx/models/message/mentions.py [hunk 4]
```diff
@@ -15,53 +15,123 @@
 )
 
 
+def build_embed_mention(
+    mention_type: MentionTypes,
+    entity: Optional[UUID],
+    name: Optional[str],
+) -> str:
+    name = name or ""
+    entity_id = entity or ""
```
Comment (Nimond) [inline pybotx/models/message/mentions.py:24]: `entity_id_str = str(entity_id) or ""`

File: pybotx/models/message/mentions.py [hunk 5]
```diff
@@ -15,53 +15,123 @@
 )
 
 
+def build_embed_mention(
+    mention_type: MentionTypes,
+    entity: Optional[UUID],
```
Comment (Nimond) [inline pybotx/models/message/mentions.py:20]: `entity_id`

File: pybotx/models/message/mentions.py (hunk 4 above)
Comment (kutuzov13) [reply pybotx/models/message/mentions.py]: Fix

**Review by Nimond — APPROVED** [review]

**Review by lxmnk — APPROVED** [review]

**Pull request merged**